test/compile/data/bin.test.ts

```diff
@@ -12,53 +12,142 @@ describe('compile/data/bin', function() {
     describe('binned field with custom extent', () => {
       it('should add bin transform and correctly apply bin', function() {
         const model = parseUnitModel({
-          mark: "point",
+          mark: 'point',
           encoding: {
             y: {
               bin: {extent: [0, 100]},
               'field': 'Acceleration',
-              'type': "quantitative"
+              'type': 'quantitative'
             }
           }
         });
         const transform = vals(bin.parseUnit(model))[0];
         assert.deepEqual(transform[0], {
           type: 'bin',
           field: 'Acceleration',
-          as: ['bin_Acceleration_start', 'bin_Acceleration_end'],
+          as: ['bin_extent_0,100_maxbins_10_Acceleration_start', 'bin_extent_0,100_maxbins_10_Acceleration_end'],
           maxbins: 10,
           extent: [0, 100],
-          signal: "Acceleration_bins",
+          signal: 'bin_extent_0_100_maxbins_10_Acceleration_bins',
         });
       });
     });
 
     describe('binned field without custom extent', () => {
       const model = parseUnitModel({
-        mark: "point",
+        mark: 'point',
         encoding: {
           y: {
             bin: true,
             'field': 'Acceleration',
-            'type': "quantitative"
+            'type': 'quantitative'
           }
         }
       });
       const transform = vals(bin.parseUnit(model))[0];
 
       it('should add bin transform and correctly apply bin', function() {
+        assert.deepEqual(transform.length, 2);
         assert.deepEqual(transform[0], {
           type: 'extent',
           field: 'Acceleration',
-          signal: 'Acceleration_extent'
+          signal: 'bin_maxbins_10_Acceleration_extent'
         });
         assert.deepEqual(transform[1], {
           type: 'bin',
           field: 'Acceleration',
-          as: ['bin_Acceleration_start', 'bin_Acceleration_end'],
+          as: ['bin_maxbins_10_Acceleration_start', 'bin_maxbins_10_Acceleration_end'],
           maxbins: 10,
-          signal: "Acceleration_bins",
-          extent: {signal: 'Acceleration_extent'}
+          signal: 'bin_maxbins_10_Acceleration_bins',
+          extent: {signal: 'bin_maxbins_10_Acceleration_extent'}
+        });
+      });
+    });
+    describe('same field encoded twice', () => {
```

Comment:
To remind, `describe` should be for grouping test suites (for namespaces and methods). 
`it` = test cases.